### ISSUE TITLE
AWS Integration 1.18.2 - Unable to collect most logs and metrics

### ISSUE THREAD

**Issue by jamesspi** [elastic/integrations#4052]
As described on [slack](https://elastic.slack.com/archives/CJZ9U5G9X/p1661200461110369), I've noticed some inconsistencies between the AWS 1.17.1 integration using v8.3.2 of Agent, and v1.18.2 and 8.4 of agent. Same AWS credentials and account, but I am unable to collect anything on the 8.4 cluster other than ECS metrics.

The following error is present with the 8.4 cluster:

```
[elastic_agent][error] 2022-08-22T21:56:46Z - message: Application: metricbeat--8.4.0[2818a0dc-3797-4b0c-b21c-ed99225c619d]: State changed to FAILED: 1 error occurred:
	* 18 errors: Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 95d95d40-14ce-47e1-b1e0-ad783282a570, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: f1319877-7ba1-46ed-9bc9-08cdc6a56145, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: fc67be11-b5bf-4d66-a96c-367c69f7a5e8, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 55855321-d861-4c01-984d-814e95a5e2a3, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: b4b4e819-590a-48d6-ae6e-fc4aa3e0d7c5, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 44730273-8658-42d1-a1c4-95f84a266f11, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: a2e0120a-19ce-49c6-b099-028c8640d3d2, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 32b53619-c4ef-40ff-9873-5d9cd256c177, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 9cebeb3b-34a9-48ff-abd5-9486b7d4fb6e, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 9dffe7ca-80b2-4cd9-a79a-62ce75bcee4b, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: e89e191a-6af6-4190-af6a-dc2af002c7fe, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 4b05a661-61b5-4e0b-88f4-26008b6d04aa, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: b7160576-cbc6-4722-a807-ef9b4a8e029e, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 3c2477b6-6bea-4df1-9fb4-b747b6398cf8, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: b9eaae8b-0d50-48e6-b62b-0da24cfb0eb1, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: db7227c1-b8a1-4e8c-a100-c6baa337f476, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 34af4b4f-77a2-49b1-8233-60342debf149, api error AuthFailure: AWS was not able to validate the provided access credentials; Error creating runner from config: 1 error: error creating aws metricset: failed DescribeRegions: operation error EC2: DescribeRegions, https response error StatusCode: 401, RequestID: 8c301732-51c1-4204-84d1-66e59dd5ec3e, api error AuthFailure: AWS was not able to validate the provided access credentials

 - type: 'ERROR' - sub_type: 'FAILED'
 ```
 
 No errors or issues on 8.3.
 
 Below are two screenshots of the resulting data streams:
 
 8.3:
 
![image](https://user-images.githubusercontent.com/20421688/186026990-66cc93a0-0ade-49e0-96ea-22f8445bf57e.png)

 8.4:
 
![image](https://user-images.githubusercontent.com/20421688/186027070-4de245df-2b9c-4692-9a4c-b289b09aa37a.png)

**Comment by sayden** [comment]
Hey @jamesspi :slightly_smiling_face: 

I spent some hours with this. Overall my local tests worked, I was able to get data from many services using a 1.8.2 barebones configuration. The two policies we have seen (the one working and the one not working) are radically different. One is ~2000 lines and the other is ~3000 lines.

I am not assuming an error generating the policies, but definitely there's something in between those two that is producing the error. Maybe you can try a diff to see if you can ring some bell knowing the context? I did the diff by myself, but the differences are massive. I have seen stuff like having `endpoint: amazonaws.com` or having `tags_filter: null` here and there that might be affecting the overall result.

Could the customer, if any :sweat_smile: try small incrementals in their policy. I'm afraid that the there's one small thing that Agent / Fleet is passing wrongly to Metricbeat config and that is why is failing, but overall I don't see anything wrong.

So either:
* The config passed through the wire is not correct, as expected looking at the policy.
* The config is correct, and some very specific field is causing the trouble. Nothing that we are aware of, though.

How does it sound? :slightly_smiling_face:

**Comment by jamesspi** [comment]
Hey @sayden , I'm the customer/user in this case.

If there are differences in the policy, it's totally down to any hidden/default parameters that the user wouldn't see/edit by default. Also, if that is indeed the case, the authentication errors we are returning in the agent log are very misleading. 

I'll try something more incremental and see how it goes.

**Comment by aspacca** [comment]
@jamesspi 

we fixed a problem with aws integrations and 8.4 in https://github.com/elastic/integrations/pull/4103

can you ask the customer to upgrade the packages to 1.23.4 and see if this will solve the issue?

**Comment by jamesspi** [comment]
Thanks, will try that ASAP.
